test/integration/collection-management/collection.test.ts

```diff
@@ -26,90 +26,38 @@ describe('Collection', function () {
-          });
-          expect(found).to.be.true;
-          // Rename the collection and check that it's gone
-          db.renameCollection('test_collection_methods', 'test_collection_methods2', err => {
-            expect(err).to.not.exist;
-            // Drop the collection and check that it's gone
-            db.dropCollection('test_collection_methods2', (err, result) => {
-              expect(result).to.be.true;
-            });
-
-            db.createCollection('test_collection_methods3', (err, collection) => {
-              // Verify that all the result are correct coming back (should contain the value ok)
-              expect(collection.collectionName).to.equal('test_collection_methods3');
-
-              db.createCollection('test_collection_methods4', (err, collection) => {
-                // Verify that all the result are correct coming back (should contain the value ok)
-                expect(collection.collectionName).to.equal('test_collection_methods4');
-                // Rename the collection and with the dropTarget boolean, and check to make sure only onen exists.
-                db.renameCollection(
-                  'test_collection_methods4',
-                  'test_collection_methods3',
-                  { dropTarget: true },
-                  err => {
-                    expect(err).to.not.exist;
-
-                    db.dropCollection('test_collection_methods3', (err, result) => {
-                      expect(result).to.be.true;
-                      done();
-                    });
-                  }
-                );
-              });
-            });
-          });
-        });
-      });
+    it('should correctly execute basic collection methods', async function () {
+      const collection = await db.createCollection('test_collection_methods');
+      // Verify that all the result are correct coming back (should contain the value ok)
+      expect(collection.collectionName).to.equal('test_collection_methods');
+      // Let's check that the collection was created correctly
+      const documents = await db.listCollections().toArray();
+      expect(documents).to.be.an('array');
+      expect(documents).to.have.length(1);
+      const nameArray = documents.map(doc => doc.name);
+      expect(nameArray).to.include('test_collection_methods');
+      // Rename the collection and check that it's gone
+      const renameCollection = await db.renameCollection(
+        'test_collection_methods',
+        'test_collection_methods2'
+      );
+      expect(renameCollection.collectionName).to.equal('test_collection_methods2');
+      // Drop the collection and check that it's gone
+      const drop = await db.dropCollection('test_collection_methods2');
+      expect(drop).to.be.true;
     });
 
-    it('should correctly access collection names', function (done) {
+    it('should correctly access collection names', async function () {
```

Comment:
With this title and the other one, they are a lot more descriptive but I think you can be more specific about the intention of the test. Take this part `access [...] collection namespace from emptyDb.collections()` when reading that title what does `emptyDb` mean to someone who has not read the contents of the test? We want titles that describe the goal of the coverage / intention of the code:

```md
- context: when collections() is called on a database with dotted collection names
  - it: returns an array
  - it: returns an array of Collections
  - it: returns an array of Collections where each .collectionName includes the dot
  - it: returns an array of Collections where each .namespace includes the db name and dot
```

I'm not asking for more break down of the test, but consider the structure above as an idea of what to try and describe in a title, if a test starts failing, the failing assertion tells us what value is unexpected, the title should answer the question "What should the value be? and what should the API under test be accomplishing?"

cc: @baileympearson if you have more advice to share, please chime in :)

Reply:
A useful mental exercise I use is "given, action, outcome" (not very catchy but it works for me 🤷 ).  

- given a collection object
- when we call list collections with a dotted collection name ('foo.bar')
- the driver should create a collection whose name is the full collection name, including dots ('foo.bar')

I like this mental model because it applies to unit tests and integration tests nicely.  For unit tests, it could be something like "given an instance of CommandOperation, when we call execute(), we expect server.commandAsync() to be called exactly once".  For integration tests, it could be like "given a collection and a 7.0+ atlas replica set and there are search indexes on the collection, when I run a `listSearchIndexes` command, I get back an array of all the search indexes".

I find that thinking this way, specifically thinking about actions / outcomes, maps nicely to good test descriptions.

```typescript
context(`when the name of the collection has a dot in it ('foo.bar')`, async function() {
  it(`creates a collection with the name as specified, including dots`, () => { ... })
});
```